mailpoet/assets/js/src/automation/editor/components/panel/step-name.tsx

```diff
@@ -0,0 +1,52 @@
+import { Dropdown, TextControl } from '@wordpress/components';
+import { edit, Icon } from '@wordpress/icons';
+import { dispatch, useSelect } from '@wordpress/data';
+import { PlainBodyTitle } from './plain-body-title';
+import { TitleActionButton } from './title-action-button';
+import { store } from '../../store';
+import { Step } from '../workflow/types';
+
+type Props = {
+  step: Step;
+};
+export function StepName({ step }: Props): JSX.Element {
+  const { stepType } = useSelect(
+    (select) => ({
+      stepType: select(store).getStepType(step.key),
+    }),
+    [],
+  );
+  return (
+    <Dropdown
+      className="mailpoet-step-name-dropdown"
+      position="bottom center"
+      renderToggle={({ isOpen, onToggle }) => (
+        <PlainBodyTitle
+          title={step.name && step.name.length > 0 ? step.name : stepType.title}
+        >
+          <TitleActionButton onClick={onToggle} aria-expanded={isOpen}>
```

Comment:
If we have `aria-expanded` I think it would also be appropriate to set an [aria-controls](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded#buttons), otherwise I don't think it's clear to assistive technology what is actually open.

Reply:
To be honest, I am unsure of this. We can not really use `aria-controls` as the ID we could use is not yet in the DOM. Using `Tab` for navigation and screenreader: The behavior of `DropDown` is such, that the component sets the focus inside the DropDown, so `aria-controls` seems not really needed. Not sure about `aria-expanded`. It feels a bit useless as in the moment I this state changes to `true` I am inside the DropDown and moving outside (e.g. to focus again on the toggle button) means, the state is again `false`. But it is [mentioned of the documentation](https://developer.wordpress.org/block-editor/reference-guides/components/dropdown/) and we use it also here https://github.com/mailpoet/mailpoet/blob/trunk/mailpoet/assets/js/src/automation/editor/components/header/document_actions.tsx#L85. So I would rather opt for it to stay. 

I asked [internally in the #a11y slack](https://a8c.slack.com/archives/C0FGK24F9/p1660283591739749), maybe they know a bit more.

Reply:
Thanks for the explanation and for asking the question in slack! As of right now it looks like no one has responded. That being said, leaving it as-is makes sense to me for consistency if nothing else, and we can revisit if we find out this is an actual accessibility problem, which seems unlikely.